ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -30,15 +30,31 @@
 
 #include <fstream>
 
+namespace gcs = google::cloud::storage;
+
 gcp_connection::gcp_connection(const std::string &bucket_name)
     : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name)
 {
 }
 
 // Builds a list of object names from the bucket.
 int
-gcp_connection::list_objects(std::vector<std::string> &objects) const
+gcp_connection::list_objects(std::vector<std::string> &objects, bool list_single) const
 {
+    using namespace gcs;
+    gcs::Client client = _gcp_client;
+    std::string const bucketName = gcp_connection::_bucket_name;
+    
+    for (auto&& object_metadata : client.ListObjects(bucketName)) {
+        if (!object_metadata) throw std::move(object_metadata).status();
```

Comment:
What does this do? Please add a comment describing this. Ideally I do not want to do throw exceptions here.

Reply:
Comments have been added here: [600b09f](https://github.com/wiredtiger/wiredtiger/pull/8717/commits/600b09f6e6320c6b3a271bf48c7bbe49933a1788) throw exception is removed and it will print a message in the terminal when it encounters an error here, WT error code will be mapped later